### PR TITLE
qt: Assimp dependency specification is more nuanced

### PR DESCRIPTION
**Pull request by chuckatkins**
libassimp has been a dependency for all of 5.x but expressing that has
varied significantly throughout the 5.x lifecycle:

  v5.0:  qt3d uses internal-only libassimp
  v5.5:  external-only libassimp
  v5.6:  either internal or external libassimp via autodetection
  v5.9:  user-selectable internal-vs-external via -assimp
  v5.14: additional qtquick3d module uses -assimp
  v5.15: qtquick3d switches to the -quick3d-assimp option
    * current bug where the incorrect target is setup

### CODE REVIEW

File: var/spack/repos/builtin/packages/qt/package.py [hunk 1]
```diff
@@ -724,17 +718,28 @@ def configure(self, spec, prefix):
...
-                    '-skip', 'webglplugin',
-                    '-skip', 'qt3d',
-                ])
+            config_args.extend(['-skip', 'qt3d'])
+
+            if '@5.10:' in spec:
+                config_args.extend(['-skip','webglplugin'])
 
-            if version >= Version('5.14'):
+            if '@5.14:' in spec:
```
Comment (sethrj) [inline var/spack/repos/builtin/packages/qt/package.py]: If you're going to change how version comparison is written in some parts of a function unrelated to the change set, maybe update the rest of the version comparisons in the function for consistency?

Comment (chuckatkins) [reply var/spack/repos/builtin/packages/qt/package.py]: That's fair.  It was already mixed in different parts of the package.  I'll try to make the adjustments consistent with existing formatting

**Comment by chuckatkins** [comment]
I've still got this marked as "dont-merge-yet" until I enable the necessary CI pieces

**Comment by wdconinc** [comment]
From 3a4d4b51c1df6856bfc4b37abd5a7f830c24accd I could build without issues the following versions, which all pull in the same assimp:
```yaml
spack:
  specs:
  - qt@5.15.3~dbus~debug~doc~examples~framework~gtk+gui+opengl~phonon+shared+sql+ssl+tools~webkit
    ^mesa+glx~llvm+opengl+opengles+osmesa~strip
  - qt@5.14.2~dbus~debug~doc~examples~framework~gtk+gui+opengl~phonon+shared+sql+ssl+tools~webkit
    ^mesa+glx~llvm+opengl+opengles+osmesa~strip
  - qt@5.12.10~dbus~debug~doc~examples~framework~gtk+gui+opengl~phonon+shared+sql+ssl+tools~webkit
    ^mesa+glx~llvm+opengl+opengles+osmesa~strip
  packages:
    all:
      compiler: [gcc@11.3.0]
```
For versions qt@5.15.3 and qt@5.14.2 I can see that `ldd $(spack location -i qt)/plugins/assetimporters/libassimp.so` indeed links against the external assimp. For version qt@5.12.10 I can see that `ldd $(spack location -i qt)/plugins/sceneparsers/libassimpsceneimport.so` indeed links against the external assimp.

I did not try to build older versions of qt (as they take a while).

**Review by sethrj — APPROVED** [review]

File: var/spack/repos/builtin/packages/qt/package.py [hunk 2]
```diff
@@ -585,12 +588,6 @@ def use_spack_dep(spack_pkg, qt_name=None):
                 '-no-nis',
             ])
 
-        if '+opengl' in spec:
-            if version >= Version('5.14'):
-                use_spack_dep('assimp')
```
Comment (sethrj) [inline var/spack/repos/builtin/packages/qt/package.py:590]: Yeah somehow this got moved into the `~opengl` branch?????

**Pull request merged**